### DISCUSSION TITLE
How to integrate with flutter?

### DISCUSSION THREAD

**Discussion by dukesx** [nextauthjs/next-auth#1534] [Help]
I am creating a flutter app where I need to authenticate social/oAuth login or sign ups. 

Now, since I am using next auth, there is no client for flutter to work with and I don't see how I can sign in or sign up using rest api exposed by next auth and get my jwt token issued. 

Let's say I can register using rest api exposed by next auth  but what about getting a signed token? And how do I tell next auth that Google auth client  says OK so just say OK and issue the user a token. 
 
Can anyone help? I am really stuck here.

**Comment by md186** [comment]
Im having the same problem. I’m having a next app with api routes and I’m doing authentification with nextauth. I’m wondering how to call register/signin by getting the jwt to have a valid session.

searching the Internet i didn’t find any clue on how to do this. Please help!

**Comment by DarkSmile92** [comment]
Anyone had luck solving this? Finding myself in the same situation...

**Comment by pierre-H** [comment]
See https://github.com/nextauthjs/next-auth/issues/1110

**Comment by GAMEA9G** [comment]
please help

**Comment by GAMEA9G** [comment]
google does not give client secret for android or ios authetication Thats one of the huge problem

one way to solve the issue for that is to build a second client using next.js and use webview to sign in . This has a advantage that you only need one client id and client secret for all platforms
